jupyterhub/auth.py

```diff
@@ -282,6 +299,14 @@ def check_blacklist(self, username):
             return
 
         if whitelist_pass:
+            # Have to merge any existing admin status returned by authenticate with is_admin
+            current_admin_status = authenticated['admin']
+            is_admin_status = await maybe_future(self.is_admin(handler, data, authenticated))
+            if current_admin_status is None:  # use whatever is_admin gave
+                authenticated['admin'] = is_admin_status
+            elif is_admin_status is not None:  # Merge Bool results from both
+                authenticated['admin'] = current_admin_status or is_admin_status
```

Comment:
Admin status configuration has gotten pretty ugly, but I believe this is the correct logic for merging anything that comes out of `authenticate`.

Reply:
Let's simplify:

```python
if authenticated['admin'] is None:
    authenticated['admin']  = await self.is_admin(handler, authenticated)
```

Basically, there's no reason to call or implement `is_admin` if authenticate is returning the admin status. Providing admin status from two sources should be considered mutually exclusive.